src/Tribe/Aggregator/Cron.php

```diff
@@ -306,16 +323,73 @@ public function verify_fetching_from_service() {
 		) );
 
 		if ( ! $query->have_posts() ) {
+			$this->log( 'debug', 'No Records Pending, skipped Fetching from service' );
 			return false;
 		}
 
 		foreach ( $query->posts as $post ) {
-			$record = Tribe__Events__Aggregator__Records::instance()->get_by_post_id( $post );
+			$record = $records->get_by_post_id( $post );
+
+			// Just double Check for CSV
 			if ( 'csv' === $record->origin ) {
+				$this->log( 'debug', sprintf( 'Record (%d) skipped, has CSV origin', $record->id ) );
 				continue;
 			}
 
-			$record->process_posts();
+			// Open a Queue to try to process the posts
+			$queue = $record->process_posts();
+
+			if ( ! is_wp_error( $queue ) ) {
+				$this->log( 'debug', sprintf( 'Record (%d) has processed queue ', $queue->record->id ) );
+				$activity = $queue->activity->get();
+				foreach ( $activity as $key => $actions ) {
+					foreach ( $actions as $action => $ids ) {
+						if ( empty( $ids ) ) {
+							continue;
+						}
+						$this->log( 'debug', sprintf( "\t" . '%s — %s: %s', $key, $action, implode( ', ', $ids ) ) );
+					}
+				}
+			} else {
+				$this->log( 'debug', sprintf( 'Record (%d) — %s', $record->id, $queue->get_error_message() ) );
+			}
+		}
+	}
+
+	/**
+	 * Allows us to log if we are using WP_CLI to fire this cron task
+	 *
+	 * @see    http://wp-cli.org/docs/internal-api/#output
+	 *
+	 * @param  string $type    What kind of log is this
+	 * @param  string $message message displayed
+	 *
+	 * @return void
+	 */
+	public function log( $type = 'colorize', $message = '' ) {
+		// Only do something When in WP_CLI
+		if ( ! class_exists( 'WP_CLI' ) ) {
```

Comment:
If we're not in a WP_CLI environment, should we fall back on default logging? ie:

`Tribe__Main::instance()->log()->log_debug( 'Msg', 'src' );`